### ISSUE TITLE
Add regime classification module (HMM + ADX/ATR + session rules + PELT change point detection)

### ISSUE THREAD

**Issue by MiledC** [MiledC/fxer#1]
## Context

The regime classifier is the critical gate between signal generation and execution. `project.md` calls this "arguably the most underappreciated component" — a trend-following strategy achieving 60% win rate in trending markets will devastate an account during ranging periods, and gold spends ~40% of its time in consolidation.

**No `fxer/regime/` directory exists today.** The `SignalGenerator` currently publishes every non-neutral signal directly to the EventBus with zero regime filtering.

## Requirements

- [ ] New package `src/fxer/regime/` with `__init__.py`
- [ ] `regime/types.py` — `RegimeState` enum (`LOW_VOL_TREND`, `HIGH_VOL_TREND`, `RANGING`), `RegimeDecision` dataclass (state, confidence, position_multiplier, should_trade, reason), `RegimeEvent` for EventBus
- [ ] `regime/hmm.py` — 2-3 state Gaussian HMM
  - Uses `hmmlearn.GaussianHMM` with `covariance_type="full"`, `n_iter=1000`
  - Input features: daily returns, volatility (rolling std), daily range
  - Methods: `fit(daily_df)`, `predict_regime(daily_df) -> RegimeState`, `save(path)`, `load(path)`
  - Returns both the state and its posterior probability
- [ ] `regime/intraday_filter.py` — Real-time ADX/ATR-based intraday classification
  - ADX < 20 → ranging (mean-reversion strategies), ADX > 25 → trending (momentum strategies)
  - Rapidly expanding ATR detection for volatility breakouts
  - Requires new `ADX` indicator class in `features/technical/indicators.py` (same pattern as RSI, MACD, etc.)
- [ ] `regime/session_rules.py` — Session-aware regime rules
  - Asian (22:00-07:00 UTC) → consolidation expected
  - London open (07:00-08:00 UTC) → breakout expected
  - London-NY overlap (12:00-16:00 UTC) → directional moves
  - Late NY (after 17:00 UTC) → avoid new entries
  - Leverages existing `TradingSession` definitions in `config/constants.py`
- [ ] `regime/change_point.py` — PELT change point detection
  - Uses `ruptures.Pelt` with O(n) complexity
  - Detects structural breaks in returns/volatility series
- [ ] `regime/classifier.py` — Composite `RegimeClassifier` combining all sub-components
  - `classify(bar, features, daily_bars) -> RegimeDecision`
  - Position sizing multipliers: 1.5x low-vol, 1.0x normal, 0.5x high-vol
  - `should_trade = False` when confidence < 0.60
- [ ] Integration: gate signals in `SignalGenerator._on_feature_event()` before publishing
- [ ] New `RegimeClassificationError` exception in `core/exceptions.py`
- [ ] Settings: `regime_hmm_states`, `regime_adx_trend_threshold`, `regime_adx_range_threshold`, `regime_confidence_minimum`, `regime_model_dir`
- [ ] Add `hmmlearn>=0.3.0` and `ruptures>=1.1.0` to `pyproject.toml`
- [ ] Unit tests in `tests/unit/test_regime.py`

## Key Technical Details

```python
# From project.md:
from hmmlearn.hmm import GaussianHMM
returns = np.column_stack([daily_returns, volatility, range_metric])
model = GaussianHMM(n_components=3, covariance_type="full", n_iter=1000)
model.fit(returns)
current_regime = model.predict(returns)[-1]
```

ADX indicator must follow the existing pattern in `indicators.py` — class with `__init__`, `update()`, `ready` property, `reset()`, `compute_batch()`.

## Files

**New:**
- `src/fxer/regime/__init__.py`
- `src/fxer/regime/types.py`
- `src/fxer/regime/hmm.py`
- `src/fxer/regime/intraday_filter.py`
- `src/fxer/regime/session_rules.py`
- `src/fxer/regime/change_point.py`
- `src/fxer/regime/classifier.py`
- `tests/unit/test_regime.py`

**Modified:**
- `src/fxer/signals/generator.py` — inject RegimeClassifier, gate signal publishing
- `src/fxer/features/technical/indicators.py` — add ADX class
- `src/fxer/config/constants.py` — ADX params and warmup
- `src/fxer/config/settings.py` — regime settings
- `src/fxer/core/exceptions.py` — RegimeClassificationError
- `pyproject.toml` — hmmlearn, ruptures deps

## Dependencies
Issues 4 and 6 also add filters to `SignalGenerator`. All three filters should be applied in sequence: economic event (cheapest) → MTF trend → regime.
